hpx/lcos/local/spinlock_pool.hpp

```diff
@@ -14,39 +14,46 @@
 #define HPX_LCOS_LOCAL_SPINLOCK_POOL_HPP
 
 #include <hpx/config.hpp>
+#include <hpx/util/fibhash.hpp>
 #include <hpx/lcos/local/spinlock.hpp>
 
 #include <cstddef>
 
 ///////////////////////////////////////////////////////////////////////////////
 namespace hpx { namespace lcos { namespace local
 {
-#if HPX_HAVE_ITTNOTIFY != 0
     namespace detail
     {
-        template <typename Tag>
+#if HPX_HAVE_ITTNOTIFY != 0
+        template <typename Tag, std::size_t N>
         struct itt_spinlock_init
         {
             itt_spinlock_init();
             ~itt_spinlock_init();
         };
-    }
 #endif
+        // special struct to ensure cache line alignemnt
+        struct spinlock_holder
```

Comment:
Couldn't `spinlock` have `alignas(64)` instead?

Reply:
Yes, I wasn't sure if that was working on all our platforms... I remember that some had problems with over alignment